packages/opensrp-plans/src/helpers/dataLoaders.ts

```diff
@@ -0,0 +1,43 @@
+/** get the full product Catalogue
+ */
+
+import { store, makeAPIStateSelector } from '@opensrp/store';
+import { OpenSRPService as GenericOpenSRPService } from '@opensrp/server-service';
+import { OPENSRP_API_BASE_URL, OPENSRP_PLANS } from '../constants';
+import { fetchPlanDefinitions } from '../ducks';
+import { PlanDefinition } from '@opensrp/plan-form-core';
+
+/** OpenSRP service */
+export class OpenSRPService extends GenericOpenSRPService {
+  constructor(endpoint: string, baseURL: string = OPENSRP_API_BASE_URL) {
+    const sessionSelector = makeAPIStateSelector();
```

Comment:
The call to the selector factory can be moved outside the constructor for proper caching